pulsar-common/src/main/java/org/apache/pulsar/common/util/keystoretls/KeyStoreSSLContext.java

```diff
@@ -105,7 +105,11 @@ public KeyStoreSSLContext(Mode mode,
                 ? DEFAULT_KEYSTORE_TYPE
                 : trustStoreTypeString;
         this.trustStorePath = trustStorePath;
-        this.trustStorePassword = trustStorePassword;
+        if (trustStorePassword == null) {
+            this.trustStorePassword = "";
+        } else {
+            this.trustStorePassword = trustStorePassword;
+        }
```

Comment:
Since `null` is the default, this is effectively changing the default value. I think it'd be more transparent to update the default value.

Reply:
@michaeljmarshall I can't find a way to change the default value which in the config file to `""`, this will cause the inconsistent between config file and code. Could you give me some advice? :)

Reply:
I might be wrong, but I think it is set here:

https://github.com/apache/pulsar/blob/6657a6c2fb23be579e74bac7c7c07a70cabfb4b2/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java#L2390-L2394

It looks like that file has several variables that end in `TlsTrustStorePassword` though, so I'm not sure how they relate to this code block. I wonder if we should consider updating all of the defaults to make sure that every one supports an empty string default and is therefore consistent.

(I haven't looked closely at this part of the code base, so please take that into consideration when looking at these other variables.)

Reply:
@michaeljmarshall we have a unit test to ensure java code default value equals to config file. The config file can't have a empty value semantic, it only has  `null` or `non-empty`

Reply:
I didn't know about those tests. Thank you for explaining. In that case, I think we should document the appropriate configuration values to indicate that `null` is converted to `""`.